src/xdocs/config_coding.xml

```diff
@@ -3153,13 +3153,42 @@ public void foo(int i, String s) {}
         </p>
       </subsection>
 
+      <subsection name="Properties">
+          <table>
+              <tr>
+                  <th>name</th>
+                  <th>description</th>
+                  <th>type</th>
+                  <th>default value</th>
+              </tr>
+              <tr>
+                  <td>ignoreMismatch</td>
+                  <td>Whether to ignore mismatch in package name and directory name.</td>
+                  <td><a href="property_types.html#boolean">Boolean</a></td>
+                  <td><code>true</code></td>
+              </tr>
+          </table>
+      </subsection>
+
       <subsection name="Examples">
         <p>
           To configure the check:
         </p>
         <source>
 &lt;module name=&quot;PackageDeclaration&quot;/&gt;
         </source>
+        <p>
+            Examples of how the check works if ignoreMismatch option is set to false:
+        </p>
+        <source>
+package com.puppycrawl.tools.checkstyle.interestingFolder.Main;
+    //Violation expected at Line 1 for Main class which is not in the package mentioned above
```

Comment:
It is not explained why this is a violation. It is not mentioned what folder this package is in as we expect none if it is in the right folder.
Also isn't it weird package name ends with `Main` and class name is `Main` too? Please make example realistic that all users can relate to.